web/modules/custom/dept_migrate/src/MigrateSupport.php

```diff
@@ -0,0 +1,269 @@
+   * Database connection.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected $d7conn;
+
+  /**
+   * Drupal\Core\Entity\EntityTypeManagerInterface definition.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * Constructs a new instance of this object.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->dbconn = Database::getConnection('default', 'default');
+    $this->d7conn = Database::getConnection('default', 'migrate');
+  }
+
+  /**
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to sync domain/groups for.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function syncDomainsToGroups(EntityInterface $entity) {
+    if ($entity instanceof NodeInterface) {
+      $this->syncNodeDomainsToGroups($entity);
+    }
+    elseif ($entity instanceof UserInterface) {
+      $this->syncUserDomainsToGroups($entity);
+    }
+    else {
+      throw new \Exception('Unhandled entity type for ' . $entity->getEntityTypeId());
+    }
+  }
+
+  /**
+   * @param string $domain_machine_name
+   *   The machine name of the D7 domain.
+   * @return int
+   *   The group id (GID) of the group entity.
+   */
+  public function domainToGroupId(string $domain_machine_name) {
+    $group_id = 0;
+
+    switch ($domain_machine_name) {
+      case 'newnigov':
+      case 'del':
+      case 'dcal':
+      case 'doe':
+        $group_id = 1;
+        break;
+
+      case 'dfp':
+        $group_id = 2;
+        break;
+
+      case 'daera':
+        $group_id = 3;
+        break;
+
+      case 'communities':
+        $group_id = 4;
+        break;
+
+      case 'economy':
+        $group_id = 5;
+        break;
+
+      case 'education':
+        $group_id = 6;
+        break;
+
+      case 'execoffice':
+        $group_id = 7;
+        break;
+
+      case 'health':
+        $group_id = 8;
+        break;
+
+      case 'infrastructure':
+        $group_id = 9;
+        break;
+
+      case 'justice':
+        $group_id = 10;
+        break;
+    }
+
+    return $group_id;
+  }
+
+  /**
+   * @param \Drupal\node\NodeInterface $node
+   *   The node object.
+   */
+  protected function syncNodeDomainsToGroups(NodeInterface $node) {
+    // Fetch the map/details of the D7 entity.
+    $d7_entity = reset(\Drupal::service('dept_migrate.migrate_uuid_lookup_manager')->lookupByDestinationNodeIds([$node->id()]));
+
+    if (!empty($d7_entity['domains'])) {
+      $plugin_id = 'group_node-' . $node->bundle();
+
+      $group_content_data = $this->dbconn->query("
```

Comment:
So this approach uses a static query to test whether a domain/group has content for this entity or not. We can do the same using Entity API, but it's actually quite verbose and a little less performant. If this code were being used for anything other than migrations, I'd probably opt to use Entity API and do something like load the group content entity, search for content in a given group entity, then add/delete a required. That's quite fiddly, this achieves the same thing and is _slightly_ faster.